### PR TITLE
Fixed some leaks: douple Py_INCREF on list creation

### PR DESCRIPTION
**Pull request by mgeplf**
* PyList_Append does an incref:
    https://github.com/python/cpython/blob/master/Objects/listobject.c#L294
  as does Py_BuildValue

### CODE REVIEW

**Comment by mgeplf** [comment]
Please run your normal test suite on this, if you have more.

File: efel/cppcore/cppcore.cpp [hunk 1]
```diff
@@ -91,23 +92,23 @@ static vector<double> PyList_to_vectordouble(PyObject* input) {
...
-  int index;
+static void PyList_from_vectordouble(vector<double> input, PyObject* output) {
+  size_t vector_size = input.size();
 
-  vector_size = input.size();
-  for (index = 0; index < vector_size; index++) {
-    PyList_Append(output, Py_BuildValue("f", input[index]));
+  for (size_t index = 0; index < vector_size; index++) {
+    PyObject *obj = Py_BuildValue("f", input[index]);
+    PyList_Append(output, obj);
```
Comment (wvangeit) [inline efel/cppcore/cppcore.cpp]: Just be sure. I hope PyList_Append applies a Py_INCREF ?

**Comment by wvangeit** [comment]
And btw, can we check for leaks through Travis ?

File: efel/cppcore/cppcore.cpp (hunk 1 above)
Comment (mgeplf) [inline efel/cppcore/cppcore.cpp]: Yeah, I have a link to the line (in app1) in the commit msg, 
PyList_Append calls it here: https://github.com/python/cpython/blob/master/Objects/listobject.c#L303

**Comment by mgeplf** [comment]
Sadly, not that I know, but I have never investigated it.

File: efel/cppcore/cppcore.cpp (hunk 1 above)
Comment (wvangeit) [inline efel/cppcore/cppcore.cpp]: Oops, didn't see that. Ok

Comment (mgeplf) [inline efel/cppcore/cppcore.cpp]: No problem, I should have made it clearer

Comment (wvangeit) [inline efel/cppcore/cppcore.cpp]: Well, the Python documentation should make this clearer :-)

**Pull request merged**